rfcs/text/0016_buildkite.md

```diff
@@ -0,0 +1,923 @@
+
+### Required
+
+#### Scalable
+
+Buildkite claims to support up to 10,000 connected agents "without breaking a sweat."
+
+We were able to connect 2,200 running agents and run a [single job with 1,800 parallel build steps](https://buildkite.com/elastic/kibana-custom/builds/8). The job ran with only about 15 seconds of total overhead (the rest of the time, the repo was being cloned, or the actual tasks were executing). We would likely never define a single job this large, but not only did it execute without any problems, the UI handles it very well.
+
+2,200 agents was the maximum that we were able to test because of quotas on our GCP account that could not easily be increased.
+
+We also created a job with 5 parallel steps, and triggered 300 parallel builds at once. The jobs executed and finished quickly, across ~1500 agents, with no issues and very little overhead. Interestingly, it seems that we were able to see the effects of our test in Buildkite's status page graphs (see below), but, from a user perspective, we were unable to notice any issues.
+
+![Status Graphs](../images/0016_buildkite_uptime.png)
+
+#### Stable
+
+So far, we have witnessed no stability issues in our testing.
+
+If Buildkite's status pages are accurate, they seem to be extremely stable, and respond quickly to issues.
+
+- [Buildkite Status](https://www.buildkitestatus.com/)
+- [Historical Uptime](https://www.buildkitestatus.com/uptime)
+- [Incident History](https://www.buildkitestatus.com/history)
+
+For agents, stability and availability will depend primarily on the infrastructure that we build and the availability of the cloud provider (GCP, primarily) running our agents. Since [we control our agents](#elastic-buildkite-agent-manager), we will be able to run agents across multiple zones, and possibly regions, in GCP for increased availability.
+
+They have a [99.95% uptime SLA](https://buildkite.com/enterprise) for Enterprise customers.
+
+#### Surfaces information intuitively
+
+The Buildkite UI is very easy to use, and works as expected. Here is some of the information surfaced for each build:
+
+- The overall status of the job, as well as which steps succeeded and failed.
+- Logs for each individual step
+- The timeline for each individual step, including how long it took Buildkite to schedule/handle the job on their end
+- Artifacts uploaded by each step
+- The entire agent/job configuration at the time the step executed, expressed as environment variables
+
+![Example Build](../images/0016_buildkite_build.png)
+
+Note that dependencies between steps are mostly not shown in the UI. See screenshot below for an example. There are several layers of dependencies between all of the steps in this pipeline. The only one that is shown is the final step (`Post All`), which executes after all steps beforehand are finished. There are some other strategies to help organize the steps (such as the new grouping functionality) if we need.
```

Comment:
_Psst_ we have a secret page you can use to visualise step deps here: https://buildkite.com/elastic/kibana/builds/84/dag

We're still polishing this so we can display it in more places, but it might help for debugging.